PrusaSlicer: update to 2.5.0.

sinetek opened the pull request:
Bump the version for PrusaSlicer to 2.5.0.
Note that this does not require the new occt version. When the version for occt is bumped, we can enable the new STEP file support. Otherwise just keep it as it was.

#### Testing the changes
- I tested the changes in this PR: YES

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc

sgn commented on srcpkgs/PrusaSlicer/patches/cereal-is-header-only.patch:
```diff
@@ -10,23 +10,24 @@ Index: PrusaSlicer-2.4.2/src/libslic3r/CMakeLists.txt
-+++ PrusaSlicer-2.4.2/src/CMakeLists.txt
+--- PrusaSlicer-2.5.0.orig/src/CMakeLists.txt
++++ PrusaSlicer-2.5.0/src/CMakeLists.txt
 @@ -125,7 +125,7 @@ if (NOT WIN32 AND NOT APPLE)
      set_target_properties(PrusaSlicer PROPERTIES OUTPUT_NAME "prusa-slicer")
  endif ()
  
 -target_link_libraries(PrusaSlicer libslic3r cereal)
 +target_link_libraries(PrusaSlicer libslic3r)
+
```
How come the patch change but the hunk indicator didn't change?

sinetek replied:
What is that? I changed the path so that the patch applies, without it it would not apply on my side because the path is 2.5.0 and not 2.4.2.

sgn commented on srcpkgs/PrusaSlicer/template:
```diff
@@ -1,12 +1,12 @@
-version=2.4.2
-revision=4
+version=2.5.0
+revision=5
```
```suggestion
revision=1
```

sinetek replied:
Why the revision going back to 1?

sgn commented on srcpkgs/PrusaSlicer/patches/boost-1.81.patch:
```diff
@@ -1,7 +1,7 @@
-Index: PrusaSlicer-2.4.2/src/slic3r/GUI/PrintHostDialogs.cpp
+Index: PrusaSlicer-2.5.0/src/slic3r/GUI/PrintHostDialogs.cpp
 ===================================================================
---- PrusaSlicer-2.4.2.orig/src/slic3r/GUI/PrintHostDialogs.cpp
-+++ PrusaSlicer-2.4.2/src/slic3r/GUI/PrintHostDialogs.cpp
+--- PrusaSlicer-2.5.0.orig/src/slic3r/GUI/PrintHostDialogs.cpp
++++ PrusaSlicer-2.5.0/src/slic3r/GUI/PrintHostDialogs.cpp
```
This patch doesn't need to be refreshed.

sinetek replied:
It does, the patch doesn't apply if i leave it like that. I might be wrong, let me try again.

sinetek replied:
ok i will revert it, it doesn't seem to matter in this context.

sinetek commented:
Hi, please see changes, I think it is ready for merge.
 - compiled on musl
 -  fixed the revision number
 - fixed the patching 'cereal', now it applies correctly.

sinetek commented:
bump.

kruceter reviewed (commented):
Squash your commits together; retain the explanation regarding `prusa-slicer.desktop`.

Another problem I see here is that you want to merge changes from your `master` branch, not the feature-related like you are supposed to.

From `CONTRIBUTING.md`:

```markdown
Using the `master` branch of your fork for contributing is also strongly discouraged.
It can cause many issues with updating your pull request (also called a PR), and having multiple PRs open at once.
```
